p8e-api/src/main/kotlin/io/provenance/engine/service/MailboxService.kt

```diff
@@ -31,35 +31,37 @@ class MailboxService(
     fun fragment(publicKey: PublicKey, env: Envelope) {
         log.info("Fragmenting env:{}", env.getUuid())
 
-        val signingKeyPair = affiliateService.getSigningKeyPair(publicKey)
-        val encryptionKeyPair = affiliateService.getEncryptionKeyPair(publicKey)
+        val signer = affiliateService.getSigner(publicKey)
+        val encryptionKeyRef = affiliateService.getEncryptionKeyRef(publicKey)
 
         val invokerPublicKey = env.contract.invoker.encryptionPublicKey.toPublicKey()
 
         // if the invoker public key does not match the encryption or signing public key, then error.
-        if (invokerPublicKey != encryptionKeyPair.public && invokerPublicKey != signingKeyPair.public ) {
-            log.error("Invoker publicKey: ${invokerPublicKey.toHex()} does not match application public key: ${encryptionKeyPair.public.toHex()}")
+        if (invokerPublicKey != encryptionKeyRef.publicKey && invokerPublicKey != signer.getPublicKey()) {
+            log.error("Invoker publicKey: ${invokerPublicKey.toHex()} does not match application public key: ${encryptionKeyRef.publicKey.toHex()}")
             return
         }
 
         val scopeOwners = env.scope.partiesList
             .filter { it.hasSigner() }
             .map { it.signer.encryptionPublicKey.toPublicKey() }
+            .map { affiliateService.getEncryptionKeyRef(it) }
```

Comment:
couldn't some of these parties/recitals be affiliates on other p8e instances (and therefore the affiliateService wouldn't be able to fetch an affiliate from our db)? I'm wondering if the getEncryptionKeyRef should be able to fall back to a KeyRef containing only the public key in that case?

Reply:
hmm ... this is a good point, ill need to circle around and fix this.

Reply:
Is the encryption key ref needed at all? Shouldn't the public key be enough.

Reply:
Yea, since we kept the encrypting the same, reverted.